abort h2 frame handling on StreamClosedError

arenevier opened the pull request:
workaround https://github.com/web-platform-tests/wpt/issues/27786

This does not fix the issue, but in practice, allows us to run ./wpt run
websockets/

arenevier commented:
@Hexcles @ricea @jgraham @stephenmcgruer

ricea reviewed (approved):
This looks very reasonable to me.

stephenmcgruer reviewed (approved):
From the failing unittests, please fix:

```
./wptserve/wptserve/server.py:490:9: F841 local variable 'e' is assigned to but never used
./wptserve/wptserve/server.py:528:9: F841 local variable 'e' is assigned to but never used
./wptserve/wptserve/server.py:531:12: E111 indentation is not a multiple of four
./wptserve/wptserve/server.py:532:12: E111 indentation is not a multiple of four
```

In terms of the actual change itself, I will rely on @ricea 's knowledge here as I know little of h2 or ws, so generally LGTM modulo fixing the lint failures.

Commit:
abort h2 frame handling on StreamClosedError

workaround https://github.com/web-platform-tests/wpt/issues/2778

This does not fix the issue, but in practice, allows us to run ./wpt run
websockets/
